File: MAP/JavaRepo - Toy language/Lab11/src/model/Expressions/Exp.java
package model.Expressions;

import exceptions.DivideByZeroException;
import exceptions.InvalidOperandException;
import exceptions.NotExistingException;
import utils.adt.MyIDictionary;
import utils.adt.MyIHeap;

public abstract class Exp {
    public abstract int eval(MyIDictionary<String, Integer> symboTable,
                             MyIHeap<Integer, Integer> heap) throws DivideByZeroException, InvalidOperandException, NotExistingException;
    public abstract String toString();
}

File: Year 1/MAP/C# - Toy language/Lab10/Toy Language/Toy Language/model/expressions/Exp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language.utils;

namespace Toy_Language.model.expressions
{
    public abstract class Exp
    {
        public abstract int Eval(IModDictionary<string, int> symbTable);
    }
}
